tests/Handler/StreamHandlerTest.php

```diff
@@ -654,4 +655,25 @@ public function testDrainsResponseAndReadsAllContentWhenContentLengthIsZero()
         $this->assertEquals('hi there... This has a lot of data!', stream_get_contents($stream));
         fclose($stream);
     }
+
+    public function testHonorsReadTimeout()
+    {
+        Server::flush();
+        $handler = new StreamHandler();
+        $response = $handler(
+            new Request('GET', Server::$url . 'guzzle-server/read-timeout'),
+            [
+                RequestOptions::READ_TIMEOUT => 1,
+                RequestOptions::STREAM => true,
+            ]
+        )->wait();
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertEquals('OK', $response->getReasonPhrase());
+        $body = $response->getBody()->detach();
+        $line = fgets($body);
+        $this->assertEquals("sleeping 60 seconds ...\n", $line);
+        $line = fgets($body);
+        $this->assertFalse($line);
+        $this->assertFalse(feof($body));
```

Comment:
According to the PHP docs, this value will be true if the stream has timed out. They don't mention any known exceptions.